cli/cmd/edges_test.go

```diff
@@ -0,0 +1,153 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/linkerd/linkerd2/controller/api/public"
+)
+
+type edgesParamsExp struct {
+	options      *edgesOptions
+	resSrc       []string
+	resDst       []string
+	resClient    []string
+	resServer    []string
+	resMsg       []string
+	resourceType string
+	file         string
+}
+
+func TestEdges(t *testing.T) {
+	// response content for SRC, DST, CLIENT, SERVER and MSG
+	var (
+		resSrc = []string{
+			"web-57b7f9db85-297dw",
+			"web-57b7f9db85-297dw",
+			"vote-bot-7466ffc7f7-5rc4l",
+		}
+		resDst = []string{
+			"emoji-646ddcc5f9-zjgs9",
+			"voting-689f845d98-rj6nz",
+			"web-57b7f9db85-297dw",
+		}
+		resClient = []string{
+			"web.emojivoto.serviceaccount.identity.linkerd.cluster.local",
+			"web.emojivoto.serviceaccount.identity.linkerd.cluster.local",
+			"default.emojivoto.serviceaccount.identity.linkerd.cluster.local",
+		}
+		resServer = []string{
+			"emoji.emojivoto.serviceaccount.identity.linkerd.cluster.local",
+			"voting.emojivoto.serviceaccount.identity.linkerd.cluster.local",
+			"web.emojivoto.serviceaccount.identity.linkerd.cluster.local",
+		}
+		resMsg = []string{"", "", ""}
```

Comment:
tioli: may as well have some messages in there to test both blank and not blank rendering?

Reply:
Thanks @rmars, currently the `edges` command is only returning meshed traffic with known identity, once it has the capacity to return requests that have no identity I will add to the test.